### PR TITLE
RESIDE-218 Centralised logging blog post

### PR DESCRIPTION
**Pull request by mwoodbri**

### CODE REVIEW

**Review by EmmaLRussell — COMMENTED** [review]
This is great - really clear. An architecture diagram would be helpful an maybe just a sentence on the motivation for adding this to your docker system..

**Comment by mwoodbri** [comment]
> This is great - really clear. An architecture diagram would be helpful an maybe just a sentence on the motivation for adding this to your docker system..

Thanks Emma. I knew I wouldn't get away without a diagram. They always take me longer than anything else but I'll try to rustle something up!

File: content/blog/logging-docker-elk.md [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+title: Aggregating logs from services deployed with Docker
+tags:
+- Docker
+- ELK
+- Logging
+---
+
+# Introduction
+
+The [Elastic (ELK) Stack](https://www.elastic.co/what-is/elk-stack) is commonly used to centralise logs from across
```
Comment (richfitz) [inline content/blog/logging-docker-elk.md]: Something that might be useful here is some motivating reason as to _why_ one might want to go through the hassle of setting up a logging server; what are we trying to make easier? This sort of starts from the point of view of already wanting one

File: content/blog/logging-docker-elk.md [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+   in [README.md](https://github.com/deviantony/docker-elk/blob/main/README.md):
+
+  ```sh
+  docker-compose up elasticsearch -d
+  docker-compose exec -T elasticsearch bin/elasticsearch-setup-passwords auto --batch
+  # Make a note of these passwords
+  docker-compose down
+  ```
+
+4. Bring up Elasticsearch and Kibana and create an index pattern:
```
Comment (richfitz) [inline content/blog/logging-docker-elk.md]: Can you define index pattern?

File: content/blog/logging-docker-elk.md [hunk 3]
```diff
@@ -0,0 +1,93 @@
+---
+author: Mark Woodbridge
+date: 2021-02-15
```
Comment (richfitz) [inline content/blog/logging-docker-elk.md]: This probably needs updating

File: content/blog/logging-docker-elk.md [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+
+- Users/roles with access to Kibana are suitably restricted
+- Your Elasticsearch and Kibana Docker volumes are backed-up
+- Sensitive information is either not logged by your applications or is shipped with suitable security (e.g.
+  [using SSL](https://www.elastic.co/guide/en/beats/filebeat/current/configuration-ssl.html)) and appropriately access
+  controlled. Obviously information such as credentials should never be logged.
+- Log messages are suitably structured, so they can be easily searched/filtered
+- Related containers/services share a common Docker Compose project name, Docker container tag etc, so that they can be
+  grouped as one unit within Elasticsearch/Kibana
+
```
Comment (richfitz) [inline content/blog/logging-docker-elk.md:114]: Can you add a link to our final repo? https://github.com/reside-ic/logs

**Review by richfitz — COMMENTED** [review]
I think this is great - a few suggestions for things that might make this more accessible, but agree a diagram would also be a nice addition

File: content/blog/logging-docker-elk.md (hunk 1 above)
Comment (mwoodbri) [reply content/blog/logging-docker-elk.md]: Done

File: content/blog/logging-docker-elk.md (hunk 2 above)
Comment (mwoodbri) [reply content/blog/logging-docker-elk.md]: Done

File: content/blog/logging-docker-elk.md (hunk 3 above)
Comment (mwoodbri) [reply content/blog/logging-docker-elk.md]: Done

File: content/blog/logging-docker-elk.md (hunk 4 above)
Comment (mwoodbri) [reply content/blog/logging-docker-elk.md:114]: Done

**Comment by mwoodbri** [comment]
> I knew I wouldn't get away without a diagram.

Done!

**Pull request merged**